src/precompiles.rs

```diff
@@ -452,30 +460,30 @@ pub fn blake2f(
     // NOTE: We may optimize this by unwraping both for loops
 
     for (i, h) in h.iter_mut().enumerate() {
-        let start = 4 + i * 8;
+        let start = BF2_STATEVEC_INIT + i * BF2_VEC_ELEM_SIZE;
         *h = u64::from_le_bytes(
-            calldata[start..start + 8]
+            calldata[start..start + BF2_VEC_ELEM_SIZE]
                 .try_into()
                 .map_err(|_| PrecompileError::InvalidCalldata)?,
         );
     }
 
     for (i, m) in m.iter_mut().enumerate() {
-        let start = 68 + i * 8;
+        let start = BF2_MSGVEC_INIT + i * BF2_VEC_ELEM_SIZE;
         *m = u64::from_le_bytes(
-            calldata[start..start + 8]
+            calldata[start..start + BF2_VEC_ELEM_SIZE]
                 .try_into()
                 .map_err(|_| PrecompileError::InvalidCalldata)?,
         );
     }
 
     t[0] = u64::from_le_bytes(
-        calldata[196..204]
+        calldata[BF2_OFFSET_COUNT_INIT..BF2_OFFSET_COUNT_INIT + BF2_VEC_ELEM_SIZE]
             .try_into()
             .map_err(|_| PrecompileError::InvalidCalldata)?,
     );
     t[1] = u64::from_le_bytes(
-        calldata[204..212]
+        calldata[BF2_OFFSET_COUNT_INIT + BF2_VEC_ELEM_SIZE * 2..BF2_BLOCK_FLAG]
```

Comment:
This should fix `blake2f` tests:
```suggestion
        calldata[BF2_OFFSET_COUNT_INIT + BF2_VEC_ELEM_SIZE..BF2_BLOCK_FLAG]
```

Reply:
Done!